### ISSUE TITLE
Auto-Brightness on iPhone 15 causes visual lag and audio glitches from "Dispatching Sound Complete"

### ISSUE THREAD

**Issue by FliplineStudios** [airsdk/Adobe-Runtime-Support#2917]
## Problem Description

On newer iPhones, when the device display's Auto-Brightness is enabled (which it is by default), there are continuous framerate hiccups as well as audio glitches, which are shown in Adobe Scout as an excessive time for "**Dispatching Sound Complete**" (often 80+ milliseconds each time).  The long time for dispatching Sound Complete is not only creating audio drops and crackles/glitches, but is also affecting the visuals and performance of apps due to the constant framerate drop when this occurs.  After a few seconds in dim lighting, the long time for Dispatching Sound Complete will usually go back to normal until it is in a brighter room again.

Confirmed to occur on both iPhone 15 and iPhone 15 Pro Max (with auto-brightness enabled) running iOS 17, though likely also affects some of the previous year's iPhone 14 models.  Does not occur on iPhone X and iPhone 11 (with or without auto-brightness enabled).  Also confirmed to be occurring at least as far back as apps built with AIR 32, so this not specific to any changes in AIR 33 or 50.  Occurring in live released apps in the App Store as well as adhoc test builds.

Oddly, "Dispatching Sound Complete" also seems to be showing up extremely frequently (multiple times per second) even when there are only a few looping 30-second SoundChannels playing which have not yet looped, and no other sounds playing.  Not sure if there are other objects that would dispatch a SOUND_COMPLETE event besides SoundChannel (like MovieClip?), or if "Dispatching Sound Complete" in Adobe Scout also relates to other things.  All sounds are included in the main SWF with linkage, to play them in the app via SoundChannel.

Related to #2642 

Adobe Scout when iPhone 15 Pro Max is in bright light, showing frequent peaks of 80+ ms from Dispatching Sound Complete:
![iphone_dispatchingsoundcomplete](https://github.com/airsdk/Adobe-Runtime-Support/assets/33292459/d45efbd6-c9a1-4b4b-80fc-9bc13bb161c1)

## Known Workarounds

Disable Auto-Brightness in Settings > Accessibility > Display & Text Size > Auto-Brightness
(though it's not an easy "fix" to communicate to end users)

**Comment by ajwfrost** [comment]
Thanks @FliplineStudios  - we hadn't spotted this metric from Scout, this could be a really good clue!

**Comment by FliplineStudios** [comment]
Some more details after making some builds with further tests:


- When no Sounds/SoundChannels are playing at all, there are no framerate hiccups affecting visuals/performance, and "Dispatching Sound Complete" does not appear in Scout.

- After reworking things so one and only one looping music track is playing, the "Dispatching Sound Complete" problem appears again, with the same spikes in framerate lag.

- Code used for playing this one looping track, playing a Sound within the SWF with linkage of "title_music":
```
var titleMusic:Sound = new title_music();
var trans:SoundTransform = new SoundTransform(1, 0);
manager.track_title = titleMusic.play(0, int.MAX_VALUE, trans);
manager.track_title.addEventListener(Event.SOUND_COMPLETE, manager.handleSoundComplete);
```

- There is no Event.SOUND_COMPLETE event being fired when listening on that single SoundChannel, even though Scout is labeling this as "Dispatching Sound Complete"

- That one looping music track would never really "complete" anyway since it's looping with int.MAX_VALUE for the number of loops

- I'm seeing a pattern with the "Dispatching Sound Complete" appearing in Scout, where for example when the game is set to 30fps it's showing up on Frames:
1303
1315
1333
1345
1363
1375
1393
1405
In this case it's looking like two "sets" that are each dispatching in 30-frame intervals, and offset from one another.  Once in a while it's a frame +/- for one of the sets.

- If I set the framerate to **40fps**, the "Dispatching Sound Complete" is appearing in two sets of **40-frame** intervals, so it seems to be related to the app's framerate.

**Comment by FliplineStudios** [comment]
Fixed in 50.2.4.3 with the audio reworking for iOS, thanks!
